prow/cmd/deck/main.go

```diff
@@ -1150,7 +1150,36 @@ func handleProwJob(prowJobClient prowv1.ProwJobInterface) http.HandlerFunc {
 	}
 }
 
-func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool) http.HandlerFunc {
+// canTriggerJob determines whether the given user can trigger any job.
+func canTriggerJob(user string, cfg config.RerunAuthConfig) bool {
+	if cfg.AllowAnyone {
+		return true
+	}
+	if cfg.AuthorizedUsers == nil {
+		return false
+	}
+	for _, s := range cfg.AuthorizedUsers {
+		if user == s {
+			return true
+		}
+	}
+	return false
+}
+
+func marshalJob(w http.ResponseWriter, pj prowapi.ProwJob) {
+	b, err := yaml.Marshal(&pj)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error marshaling: %v", err), http.StatusInternalServerError)
+		logrus.WithError(err).Error("Error marshaling jobs.")
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	if _, err := w.Write(b); err != nil {
+		logrus.WithError(err).Error("Error writing log.")
+	}
+}
+
+func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool, cfg config.Getter) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		name := r.URL.Query().Get("prowjob")
```

Comment:
nit: Please create a `logrus.Entry` with `logrus.WithField("prowjob", name)` and use that throughout so that log messages include the name of the prowjob in question (otherwise the log messages probably won't be very useful).